modules/core/src/main/scala/org/tessellation/config/types.scala

```diff
@@ -37,8 +38,9 @@ object types {
   )
 
   case class SnapshotConfig(
-    storedSnapshotPath: Path,
-    fallbackTriggerTimeout: FiniteDuration
+    globalSnapshotPath: Path,
+    fallbackTriggerTimeout: FiniteDuration,
+    inMemoryCapacity: PosInt
```

Comment:
I would use `NonNegLong` / `NonNegInt` here.

Reply:
we cannot store `0` snapshot in memory, it must be at least 1

Reply:
Why not? I think it would effectively mean turning off the in-memory cache.

Reply:
Then, we will end-up with asynchronous storage again

Reply:
Not really if the removal from the cache happens after the successful storage write.